### PR TITLE
Add various sequence diagrams to the main doc

### PR DESCRIPTION
**Pull request by madadam**
Based on the tunnel branch, so *DO NOT MERGE* until tunnel is merged.

This change is [Reviewable](https://reviewable.io/reviews/maidsafe/routing/960)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @canndrew

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/lib.rs [hunk 1]
```diff
@@ -133,6 +133,10 @@
 //! * by a client or individual node, or
 //! * by a sufficient number of members of a group authority.
 //!
+//! # Sequence diagrams
+//!
+//! [Bootstrapping](https://cacoo.com/diagrams/th3ie0I5yOM5ymJq-F56A2.png)
```
Comment (dirvine) [inline src/lib.rs]: I feel this should be live (poss not implemented version) Plus a link to a png or similar in the repo itself (img dir) that reflects the code at this point? The png will be updated with tags update and can then be referenced in the code doc further in the codebase.

**Comment by dirvine** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/lib.rs, line 138 \[r1\]](https://reviewable.io:443/reviews/maidsafe/routing/960#-KE_xha-T6rv5AAMmh46:-KE_xha-T6rv5AAMmh47:849977793) ([raw file](https://github.com/maidsafe/routing/blob/3ad6e8af22410c55b7bfeda400e002f5350c8d77/src/lib.rs#L138)):*
I feel this should be live (poss not implemented version) Plus a link to a png or similar in the repo itself (img dir) that reflects the code at this point? The png will be updated with tags update and can then be referenced in the code doc further in the codebase.
 Add a line note

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/routing/960)*

**Comment by madadam** [comment]
*[src/lib.rs, line 138 \[r1\]](https://reviewable.io:443/reviews/maidsafe/routing/960#-KE_xha-T6rv5AAMmh46:-KEa-9H_uoPHqZMwV6xF:-416578562) ([raw file](https://github.com/maidsafe/routing/blob/3ad6e8af22410c55b7bfeda400e002f5350c8d77/src/lib.rs#L138)):*
The png links should be "live" in the sense that when the diagram changes, the png should change too. At least I understood it that way.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/routing/960)*

**Comment by dirvine** [comment]
Reviewed 1 of 3 files at r3, 1 of 1 files at r4.
Review status: 2 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/lib.rs, line 138 \[r1\]](https://reviewable.io:443/reviews/maidsafe/routing/960#-KE_xha-T6rv5AAMmh46:-KEaxXVibv_iPZ5W4BdA:-651050884) ([raw file](https://github.com/maidsafe/routing/blob/3ad6e8af22410c55b7bfeda400e002f5350c8d77/src/lib.rs#L138)):*
Yes that is correct. So the live doc (on cacoo) is planned work and may not exist in code. :+1: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/routing/960)*

**Comment by dirvine** [comment]
Reviewed 1 of 3 files at r3.
Review status: 2 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/routing/960)*

**Comment by dirvine** [comment]
Reviewed 1 of 3 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/routing/960)*

**Comment by dirvine** [comment]
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/routing/960)*

**Comment by afck** [comment]
Reviewed 1 of 3 files at r3, 2 of 2 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/routing/960)*

**Pull request merged**